graphql_compiler/deserialization.py

```diff
@@ -1,5 +1,6 @@
 # Copyright 2020-present Kensho Technologies, LLC.
 """Convert values to their underlying GraphQLType."""
+from datetime import date, datetime
 from types import MappingProxyType
 from typing import Any, Callable, Mapping, Tuple, Type
```

Comment:
```suggestion
from typing import Any, Callable, Mapping, Tuple, Type, Union
```

(See my suggestion for `_parse_datetime_value`.)